app/Http/Controllers/UserImageController.php

```diff
@@ -25,15 +25,34 @@ public function store(Request $request, ImageService $imageService): RedirectRes
             if($request->user_image && $request->user_image->extension()){
 
                 $user = Auth::user();
-                // This is only needed for IDE (Don't freak out)
+                // This is only needed for IDE
                 $user = User::find($user->id);
 
                 $folder = $imageService->getUserFolder($user);
                 $uploadedFile = $request->user_image;
                 $extension = $uploadedFile->extension();
-                $filename = 'user-image' . '.' . $extension;
+                $basename = 'user-image';
+                $filename = $basename . '.' . $extension;
+
+                //Checks if there is an old image record.
+                $oldImageRecord = $user->images->first();
+
+                if($oldImageRecord){
+
+                    //Deletes Last File in Dir for User Image
+                    $oldImageDirStatus = $imageService->deleteFileAtPath($oldImageRecord->filename);
```

Comment:
move this into the if statement $imageService->deleteFileAtPath($oldImageRecord->filename)